DEVELOPER.md

```diff
@@ -520,3 +523,20 @@ In certain cases, partition may not be deleted correctly.  In this case, you can
 ```
 kubectl patch partition  <partition_name> -p '{"metadata":{"finalizers":null}}' --type merge
 ```
+
+### kube-apiserver-local: slimmed down K8s storage without docker
+
+This will run kube-apiserver without the need for Docker.
+It uses kubebrain to store the data locally and satisfy the etcd interface.
+It only works with running the binaries externally using the --local flag and additionally the --develop flag for cluster start
+
+```
+git clone https://github.com/gregwebs/kube-apiserver-local
+cd kube-apiserver-local
+./apiserver.sh
+```
+
+```
+./kubeconfig.sh
+KUBECONFIG=admin.kubeconfig kubectl ...
```

Comment:
> It looks like we only need to build the binary for arch where there is not an official kube-apiserver release, right? Does kubebrain build the binary if we are running in architecture with an official release?

Yes. I am assuming Mac users are a lot more interested in this than Linux users and K8s does not provide a darwin-arm64 binary. I should change the script to just download the binary if it is available rather than always building from source.

Reply:
Hmm, it specifically sets up a common CA.
Are you testing on Linux? I can give it a try on Linux as well and see what I get there.

Reply:
I am testing on Mac M1

Reply:
Is this the intended way to run `fluvio`?

```
 KUBECONFIG=admin.kubeconfig fluvio cluster start --local
Current channel: stable
📝 Running pre-flight checks
    ✅ Supported helm version 3.9.1+ga7c043a is installed
    ✅ Kubectl active cluster default at: https://127.0.0.1:6443 found
    ❌ Check Kubernetes version failed Failed to connect to Kubernetes via the active context
    ✅ Local Fluvio is not installed
    ❌ Check Fluvio Sys Chart failed Helm client error
💔 Some pre-flight check failed!
Error: 
   0: Fluvio cluster error
   1: Preflight check failed
```

Reply:
Oh, it looks like having `rancher desktop` running was causing troubles. Now, I don't get the certificate issue

Reply:
I ran with `--develop` as well in my testing.

Reply:
This is the whole point of running without Kubernetes 👍

Reply:
```
morenol@fischer fluvio % flvd cluster start --local --develop --log-dir ./log
    Finished dev [unoptimized + debuginfo] target(s) in 0.18s
    Finished dev [unoptimized + debuginfo] target(s) in 0.19s
     Running `target/debug/fluvio cluster start --local --develop --log-dir ./log`
📝 Running pre-flight checks
    ✅ Supported helm version 3.9.1+ga7c043a is installed
    ✅ Kubectl active cluster default at: https://127.0.0.1:6443 found
    ✅ Supported Kubernetes server 1.25.0 found
    ✅ Local Fluvio is not installed
    ✅ Fluvio system charts are installed
🎉 All checks passed!
✅ Local Cluster initialized
🖥️  Trying to connect to SC: localhost:9003 70 seconds elapsed -                                                                                                                                                                         ^C
```

I am getting timeout at that place, are you running using `fluvio cluster start --local --develop` or using `fluvio run sc --local` ?

Reply:
I used `flvd cluster start --local --develop` and then started my own sc locally myself.

Reply:
KUBECONFIG must be set for that as well.

```
KUBECONFIG=/path/to/admin.kubeconfig ./target/debug/fluvio-sc-k8 --local
```